src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -27,15 +27,33 @@
 ;; ## Plugin Initialization
 
 (defn initialize!
-  ; 1. Access app info using app accessor to determine url, app-id, etc.
-  ; 2. Create a vault client
-  ; 3. Authenticate the vault client
-  ; 4. Return the vault client
   "Set up the vault client."
   [logger app-accessor]
   (alter-var-root #'log/logger (constantly logger))
-  ;; TODO: actually initialize a Vault client
-  nil)
+  (atom nil))
+
+
+;; ## Model
+
+;; A map of user configurable fields to the all the data necessary to define those fields
+(def input-schema
+  {;; the name of the corresponding input field
+   :vault_addr {;; the metadata required by the plugin API about every input field
+                :metadata       {:required true :secure false}
+                ;; the string representation of the URL
+                :label          "Vault URL"
+                ;; an array outlining extended validation the function may require
+                :validate-funcs [{;; true if valid, false if not
```

Comment:
a more consistent clojure naming would be `validate-fns`.

Also these definitions would be more easy to make use of if they were just the full validation functions themselves, including validation error messages. Detection would be the same, where any non-nil response means a validation error.


```clojure
:validate-fns [(fn [v]
                 (when-not (or (str/starts-with? v "http://") (str/starts-with? v "https://"))
                    "Vault URL must start with http:// or https://"))]
```

This would make unpacking and using these less unwieldy since the functions could be made use of directly

At line 148 below,

```clojure
validate-func-errors (keep #(% field-value) (:validate-funcs field-model))
```

Reply:
True, I like that a lot better!